### PR TITLE
hello-rails-front-end

### PR DESCRIPTION
**Pull request by bhushan354**
- Instructions to run the project : 
 1) git checkout setup-react
 2) npm install
 3) npm start

This is Front-end of the application In which i have implemented the following things :

Create a new React app called 'hello-react-front-end'
Initialize your project with Git.
Create the App component with react-router.
Create the Greeting component that will display a greeting. Set it up as a route in your App component.
Create a store, an action and a reducer that will connect to you API endpont to get the random greeting.
Display the random greeting in your Greeting component.

Back-end here : https://github.com/bhushan354/hello-rails-back-end/pull/2

Thank you ,
Bhushan .

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,34 @@
...
+
+const App = () => {
+  const dispatch = useAppDispatch();
+  const greeting = useAppSelector((state) => state.greeting.value);
+  const loading = useAppSelector((state) => state.greeting.loading);
+  const error = useAppSelector((state) => state.greeting.error);
+
+  useEffect(() => {
+    dispatch(fetchGreeting());
+  }, []);
```
Comment (Ibizugbe) [inline src/App.js]: Nice job so far😉. As a requirement, you are to create a Greeting component that will display a greeting. You are also required to set it up as a route in your App component using react-router😉. Please do well to make adjustments👍🏼. Kindly do well to stick to the instructions as shown in the image below😉.


image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @bhushan354 👋🏼👋🏼👋🏼, 

require_changes


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Linters are passing👍🏼.
- [x] Project has an app component👍🏼.
- [x] Redux has been used properly to manage the global state of the application👍🏼.
- [x] Great UI/UX


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by bhushan354** [comment]
@Ibizugbe Sir my App.js is not rendering inside of the root component can you tell me how to do it

File: src/App.js
Comment (Efiamotu-1) [inline src/App.js:1]: - [ ] From the project requirement below you are to make use of react router in your app component, but you have not implemented this, please kindly do so :+1: 
nd

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @bhushan354, 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by bhushan354** [comment]
Hello sir ,
hope this commit fullfills all the requirements , if still there's something that you want to change please call me on zoom as this is my last review.

Thank you,
Bhushan Deshmukh.

**Review by danmainah — APPROVED** [review]
Hi @bhushan354 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented the change requested to add the Greeting component.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**